libraries/tlv-account-resolution/src/seeds.rs

```diff
@@ -0,0 +1,212 @@
+//! Types for managing seed configurations in TLV Account Resolution
+//!
+//! The largest possible seed configuration is 3 bytes (`InstructionArg`).
+//! This means that an `AccountMetaPda` can store up to 10 seed configurations.
+
+use {crate::error::AccountResolutionError, solana_program::program_error::ProgramError};
+
+/// Enum to describe a required seed for a Program-Derived Address
+#[derive(Clone, Debug, PartialEq)]
+pub enum Seed {
+    /// A literal hard-coded argument
+    Literal {
+        /// The literal value repesented as a vector of bytes.
+        /// For example, if a literal value is a string literal,
+        /// such as "my-seed", this value would be
+        /// `"my-seed".as_bytes().to_vec()`.
+        bytes: Vec<u8>,
+    },
+    /// An instruction-provided argument, to be resolved from the instruction
+    /// data
+    InstructionArg {
+        /// The index where the bytes of an instruction argument begin
+        index: u8,
+        /// The type of instruction argument to resolve
+        ty: InstructionArgType,
+    },
+    /// The public key of an account from the entire accounts list.
+    /// Note: This includes an extra accounts required.
+    AccountKey {
+        /// The index of the account in the entire accounts list
+        index: u8,
+    },
+}
+impl Seed {
+    /// Get the size of a seed configuration
+    pub fn tlv_size(&self) -> u8 {
+        match &self {
+            // 1 byte for the discriminator, 1 byte for the length of the bytes, then the raw bytes
+            Seed::Literal { bytes } => 1 + 1 + bytes.len() as u8,
+            // 1 byte for the discriminator, 1 byte for the index, 2 bytes for the type
+            Seed::InstructionArg { .. } => 1 + 1 + InstructionArgType::TLV_SIZE,
+            // 1 byte for the discriminator, 1 byte for the index
+            Seed::AccountKey { .. } => 1 + 1,
+        }
+    }
+
+    /// Packs a seed configuration into a byte vector
+    pub fn pack(&self) -> Vec<u8> {
```

Comment:
This is a good call.

At a nit-level, I'm inclined to lean more toward the naming convention in `solana_program::program_pack::Pack` and keep it as `pack`:

```rust
fn pack(src: Self, dst: &mut [u8]) -> Result<(), ProgramError>
```

Also since I have `unpack_from_array` which calls `unpack`, I can also have `pack_into_array` call `pack`. 

However, I'm totally open to `unpack_from_array` calls `unpack_from_slice`, and `pack_into_array` calls `pack_into_slice`.

Pushed changes with `pack` for now.